ecosystem/typescript/sdk/src/aptos_account.test.ts

```diff
@@ -65,3 +66,11 @@ test("Signs Strings", () => {
     "0xc5de9e40ac00b371cd83b1c197fa5b665b7449b33cd3cdd305bb78222e06a671a49625ab9aea8a039d4bb70e275768084d62b094bc1b31964f2357b7c1af7e0d",
   );
 });
+
+test("Gets the resource account address", () => {
+  const sourceAddress = "9db1d65a321e4a86f8098cb04e76fce098890b84211fa06d65f24dc644bf0fec";
+  const seedInHex = "3031" // hexadecimal form of "01" which is 3031
+  const seed = new HexString(seedInHex).toUint8Array();
+
+  expect(AptosAccount.getResourceAccountAddress(sourceAddress, seed).hex()).toBe("0x915e47f986471a5faba9cb2f726611cd554344419ddd6d5b3e9f00bcafd30169");
```

Comment:
Hey, how did you get this address in contract side?
**"0x915e47f986471a5faba9cb2f726611cd554344419ddd6d5b3e9f00bcafd30169"**

I followed your method, but I can't get proper resource account address.

Reply:
i used those seeds and address and printed the resource address created by the module using the `create_resource_address` function.

Reply:
I found the solution.
```
  const originBytes = new Uint8Array(source.length + seed.length + 1);
  originBytes.set(source);
  originBytes.set(seed, source.length);
  originBytes.set([255], source.length + seed.length);
```
You omitted to add 255 at the end of bytes.
Please look at here.
https://github.com/aptos-labs/aptos-core/blob/83641d3f083762f31700060e5e95fbb1cbc28bf0/aptos-move/framework/aptos-framework/sources/account.move#L400